### ISSUE TITLE
Gradients update primitives

### ISSUE THREAD

**Issue by stonextu** [intel/mkl-dnn#460]
Hi, 
I want to train a new net with MKL-DNN. I read through demos in examples, and found that MKL-DNN seems not to provide gradients update primitives, such as Adam etc, and as well as loss functions, such as cross entropy. Is it true that those primitives have not been implemented in MKL-DNN?

**Comment by emfomenk** [comment]
Hi @stonextu,

Yeah, that's correct. Intel MKL-DNN mostly focuses on compute intensive primitives like convolution, inner products and RNNs; and some auxiliary primitives that typically happen in-between those. The assumption is that majority of the primitives (which are not performance bottleneck) can be implemented by a user or frameworks.

In particular, the library doesn't provide any solvers or applying the weights update. For the simplest case you can you `mkldnn::sum` with scaling factor though (to implement `w = w + alpha * dw`). But of course the more advanced solvers require much more complicated operations and require manual programming.

**Comment by stonextu** [comment]
@emfomenk Thanks for your reply.
